patches/tModLoader/Terraria/Social/Base/DeveloperMetadata.cs

```diff
@@ -44,84 +44,37 @@ public class DeveloperMetadata
 {
 	public List<ModVersionHash> modVersionHashes { get; set; } = new List<ModVersionHash>();
 
-	[JsonConstructor]
-	private DeveloperMetadata() { }
-
-	internal DeveloperMetadata(string workshopPath, bool useWebApi, ModDownloadItem modDownloadItemAsFound = null)
+	internal string Serialize()
 	{
-		CalculateModHashes(workshopPath, useWebApi , modDownloadItemAsFound);
-		TrimDevMetadata();
+		return JsonConvert.SerializeObject(this, Formatting.None, new ModVersionHash.VersionHashConverter());
 	}
-	
-	internal DeveloperMetadata(string serializedDevMetadata)
+
+	internal static DeveloperMetadata Deserialize(string serializedDevMetadata)
 	{
-		// Try-Catch is for error correction for any bizarre dev metadata put on items previously
+		if (string.IsNullOrWhiteSpace(serializedDevMetadata))
+			return new();
+
 		try {
-			var devMetadata = DeSerialize(serializedDevMetadata);
+			var devMetadata = JsonConvert.DeserializeObject<DeveloperMetadata>(serializedDevMetadata, new ModVersionHash.VersionHashConverter());
 
 			if (devMetadata == null || devMetadata.modVersionHashes == null) {
```

Comment:
do we really need this guard? I'm not sure how it would be possible to hit this condition.